iks/provider/provider_test.go

```diff
@@ -106,230 +96,6 @@ func TestNewProvider(t *testing.T) {
-	defer teardown()
-
-	vpcp, err := GetTestProvider(t, logger)
-	assert.Nil(t, err)
-	// sessn, err := vpcp.OpenSession(context.Background(), provider.ContextCredentials{
-	// 	AuthType:     provider.IAMAccessToken,
-	// 	Credential:   TestProviderAccessToken,
-	// 	IAMAccountID: TestIKSAccountID,
-	// }, logger)
-
-	// require.NoError(t, err)
-	// assert.NotNil(t, sessn)
-
-	sessn, err := vpcp.OpenSession(context.Background(), provider.ContextCredentials{
-		AuthType:     provider.IAMAccessToken,
-		IAMAccountID: TestIKSAccountID,
-	}, logger)
-
-	require.Error(t, err)
-	assert.Nil(t, sessn)
-
-	sessn, err = vpcp.OpenSession(context.Background(), provider.ContextCredentials{
-		AuthType:     "WrongType",
-		IAMAccountID: TestIKSAccountID,
-	}, logger)
-
-	require.Error(t, err)
-	assert.Nil(t, sessn)
-}
-
-func TestUpdateAPIKey(t *testing.T) {
-	logger, teardown := GetTestLogger(t)
-	defer teardown()
-
-	vpcp, err := GetTestProvider(t, logger)
-	assert.Nil(t, err)
-
-	err = vpcp.UpdateAPIKey(nil, logger)
 	assert.NotNil(t, err)
-
-	config := &vpcconfig.VPCBlockConfig{
-		VPCConfig: &config.VPCProviderConfig{
-			G2APIKey: "invalid",
-			APIKey:   "invalid",
-		},
-	}
-
-	err = vpcp.UpdateAPIKey(config, logger)
-	assert.Nil(t, err)
-}
-
-func GetTestOpenSession(t *testing.T, logger *zap.Logger) (sessn *IksVpcSession, uc, sc *fakes.RegionalAPI, err error) {
-	vpcp, err := GetTestProvider(t, logger)
-	iksVpcProvider, _ := vpcp.(*IksVpcBlockProvider)
-
-	m := http.NewServeMux()
-	s := httptest.NewServer(m)
-	assert.NotNil(t, s)
-
-	//iksVpcProvider.VPCBlockProvider.httpClient = http.DefaultClient
-
-	// Inject a fake RIAAS API client
-	cp := &fakes.RegionalAPIClientProvider{}
-	uc = &fakes.RegionalAPI{}
-	cp.NewReturnsOnCall(0, uc, nil)
-	sc = &fakes.RegionalAPI{}
-	cp.NewReturnsOnCall(1, sc, nil)
-	iksVpcProvider.VPCBlockProvider.ClientProvider = cp
-
-	vpcSession := &vpcprovider.VPCSession{
-		VPCAccountID: TestIKSAccountID,
-		//Config:       iksVpcProvider.config,
-		ContextCredentials: provider.ContextCredentials{
-			AuthType:     provider.IAMAccessToken,
-			Credential:   TestProviderAccessToken,
-			IAMAccountID: TestIKSAccountID,
-		},
-		VolumeType: "vpc-block",
-		Provider:   "vpc-classic",
-		Apiclient:  uc,
-		Logger:     logger,
-		APIRetry:   vpcprovider.NewFlexyRetryDefault(),
-	}
-	sessn = &IksVpcSession{
-		VPCSession: *vpcSession,
-		IksSession: vpcSession,
-	}
-
-	return
-}
-
-func TestGetTestOpenSession(t *testing.T) {
-	//var err error
-	logger, teardown := GetTestLogger(t)
-	defer teardown()
-
-	vpcs, uc, sc, err := GetTestOpenSession(t, logger)
-	assert.NotNil(t, vpcs)
-	assert.NotNil(t, uc)
-	assert.NotNil(t, sc)
-	assert.Nil(t, err)
-
-	providerDisplayName := vpcs.GetProviderDisplayName()
-	assert.Equal(t, providerDisplayName, provider.VolumeProvider("IKS-VPC-Block"))
-	vpcs.Close()
-
-	providerName := vpcs.ProviderName()
-	assert.Equal(t, providerName, provider.VolumeProvider("IKS-VPC-Block"))
-
-	volumeType := vpcs.Type()
```

Comment:
Why we have removed this

Reply:
With the current changes, initializing provider is not possible, because it needs a k8s environment (to read secrets from which api key is read). For the time being I have removed this. Will take it up in next phase by adding fakes.